spec/integration/util/windows/user_spec.rb

```diff
@@ -117,6 +117,28 @@ def expect_logon_failure_error(&block)
       it "should return false given a nil username and an incorrect password" do
         expect(Puppet::Util::Windows::User.password_is?(nil, bad_password)).to be_falsey
       end
+
+      context "with a correct password" do
+        it "should return true even if account restrictions are in place " do
+          Puppet::Util::Windows::User.stubs(:logon_user).raises(Puppet::Util::Windows::Error.new('', 1327))
```

Comment:
A couple checks reveals...
- 1909 is indeed raised if account is locked out even when creds are valid
- in both the case that account is locked or account is expired, puppet will proceed to set the password anyway after failed logon.

Both of these (and probably others) are cases we could improve on...
I think to avoid further scope creep on this ticket perhaps its best to limit functionality to what's here, and target further improvements for https://tickets.puppetlabs.com/browse/PUP-6569 ?